pkg/controller/bastion/actuator_reconcile.go

```diff
@@ -193,10 +193,33 @@ func ensureComputeInstance(logger logr.Logger, openstackClientFactory openstackc
 		return nil, errors.New("networkInfo must not be empty")
 	}
 
+	Compute, err := openstackClientFactory.Compute()
+	if err != nil {
+		return nil, err
+	}
+
+	flavorID, err := Compute.FindFlavorID(bastionConfig.FlavorRef)
+	if err != nil {
+		return nil, err
+	}
+
+	if flavorID == "" {
+		return nil, errors.New("flavorID not found")
+	}
+
+	imageID, err := Compute.FindImagesID(bastionConfig.ImageRef)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get (create) public ip address: %w", err)
+	}
+
+	if imageID == nil {
+		return nil, errors.New("imageID not found")
+	}
+
 	createOpts := servers.CreateOpts{
 		Name:           opt.BastionInstanceName,
-		FlavorRef:      bastionConfig.FlavorRef,
-		ImageRef:       bastionConfig.ImageRef,
+		FlavorRef:      flavorID,
+		ImageRef:       imageID[0].ID,
```

Comment:
missing `len` check

Reply:
I was try like this in this morning  ` if image == nil || len(image) == 0` but it seems not passed verify 

```
  pkg/controller/bastion/actuator_reconcile.go:215:5: S1009: should omit nil check; len() for nil slices is defined as zero (gosimple)
        if image == nil || len(image) == 0 {
           ^
make: *** [check] Error 1
```

Reply:
as the recommendation says, just omit the nil check and only check for `len`

Reply:
done 0475ebe0